tests/unit/test_hnswlib.cpp

```diff
@@ -913,3 +914,51 @@ TEST_F(HNSWLibTest, hnsw_batch_iterator_advanced) {
     VecSimBatchIterator_Free(batchIterator);
     VecSimIndex_Free(index);
 }
+
+TEST_F(HNSWLibTest, hnsw_resolve_params) {
+    size_t dim = 4;
+    size_t M = 8;
+    size_t ef = 2;
+
+    VecSimParams params = {.algo = VecSimAlgo_HNSWLIB,
+                           .hnswParams = {.type = VecSimType_FLOAT32,
+                                          .dim = dim,
+                                          .metric = VecSimMetric_L2,
+                                          .initialCapacity = 0,
+                                          .M = M,
+                                          .efConstruction = ef,
+                                          .efRuntime = ef}};
+    VecSimIndex *index = VecSimIndex_New(&params);
+
+    VecSimQueryParams qparams;
+
+    VecSimRawParam *rparams = array_new<VecSimRawParam>(2);
+
+    ASSERT_TRUE(VecSimIndex_ResolveParams(index, rparams, &qparams));
```

Comment:
why this is true?

Reply:
I assume calling VecSimIndex_ResolveParams without params is OK, So empty array should return true.